File: react-native/src/service/values/wifi-security-mode.ts
/**
 * Wifi Encryption Types
 */
export const WifiSecurityModeEnum = {
  /** Undefined value */
  UNKNOWN: 'UNKNOWN',
  /** none */
  NONE: 'NONE',
  /** WEP */
  WEP: 'WEP',
  /** WPA/WPA2 PSK */
  WPA_WPA2_PSK: 'WPA_WPA2_PSK',
  /** WPA3-SAE */
  WPA3_SAE: 'WPA3_SAE',
} as const;

/** type definition of WifiSecurityModeEnum */
export type WifiSecurityModeEnum =
  typeof WifiSecurityModeEnum[keyof typeof WifiSecurityModeEnum];

File: kotlin-multiplatform/src/commonMain/kotlin/com/ricoh360/thetableclient/service/data/values/WifiSecurityMode.kt
package com.ricoh360.thetableclient.service.data.values

/**
 * Wifi Encryption Types
 */
enum class WifiSecurityMode(internal val value: String?) {
    /**
     * Undefined value
     */
    UNKNOWN(null),

    /**
     * none
     */
    NONE("none"),

    /**
     * WEP
     */
    WEP("WEP"),

    /**
     * WPA/WPA2 PSK
     */
    WPA_WPA2_PSK("WPA/WPA2 PSK"),

    /**
     * WPA3-SAE
     */
    WPA3_SAE("WPA3-SAE"),
    ;

    companion object {
        val keyName: String
            get() = "security"

        /**
         * Search by json value.
         *
         * @param value json value.
         * @return WifiSecurityMode
         */
        fun getFromValue(value: String): WifiSecurityMode {
            return entries.firstOrNull { it.value == value } ?: run {
                println("Unknown value ${UNKNOWN::class.simpleName}: $value")
                UNKNOWN
            }
        }
    }
}
